libs/wire-subsystems/src/Wire/PropertySubsystem.hs

```diff
@@ -0,0 +1,42 @@
+{-# LANGUAGE TemplateHaskell #-}
+
+module Wire.PropertySubsystem where
+
+import Data.Id
+import Data.Text.Lazy qualified as LText
+import Imports
+import Network.HTTP.Types
+import Network.Wai.Utilities qualified as Wai
+import Polysemy
+import Wire.API.Error
+import Wire.API.Error.Brig qualified as E
+import Wire.API.Properties
+import Wire.Error
+
+data PropertySubsystemError
+  = TooManyProperties
+  | PropertyKeyTooLarge
+  | PropertyValueTooLarge
+  | PropertyValueInvalid String
+  | StoredPropertyValueInvalid
+  deriving (Show, Eq)
+
+propertySubsystemErrorToHttpError :: PropertySubsystemError -> HttpError
+propertySubsystemErrorToHttpError =
+  StdError . \case
+    TooManyProperties -> errorToWai @E.TooManyProperties
+    PropertyKeyTooLarge -> errorToWai @E.PropertyKeyTooLarge
+    PropertyValueTooLarge -> errorToWai @E.PropertyValueTooLarge
+    PropertyValueInvalid err -> Wai.mkError status400 "bad-request" (LText.pack err)
+    StoredPropertyValueInvalid -> Wai.mkError status500 "internal-server-error" "Internal Server Error"
+
+data PropertySubsystem m a where
+  SetProperty :: UserId -> ConnId -> PropertyKey -> RawPropertyValue -> PropertySubsystem m ()
+  DeleteProperty :: UserId -> ConnId -> PropertyKey -> PropertySubsystem m ()
+  ClearProperties :: UserId -> ConnId -> PropertySubsystem m ()
+  OnUserDeleted :: UserId -> PropertySubsystem m ()
+  LookupProperty :: UserId -> PropertyKey -> PropertySubsystem m (Maybe RawPropertyValue)
+  GetPropertyKeys :: UserId -> PropertySubsystem m [PropertyKey]
+  GetAllProperties :: UserId -> PropertySubsystem m PropertyKeysAndValues
+
```

Comment:
Hmm, interesting idea. But I think it has two drawbacks:
1. It would be a bit confusing to understand what's going on
2. The two interfaces are similar but they could potentially change independent of each other and untangling at that point would also be a little confusing. I'd say let's keep it as it is, its not a lot of code.